File: core/src/main/java/dev/bee/kanjianki/core/SettingsImportPreset.java
package dev.bee.kanjianki.core;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public record SettingsImportPreset(
        String label,
        boolean activeCards,
        boolean suspendedCards,
        boolean taggedCards,
        String tags,
        boolean weakCards,
        double weakDifficulty,
        int weakLapses,
        int minMatchingCards,
        boolean browserQueryCards,
        String browserQuery
) {
    private static final List<SettingsImportPreset> DEFAULTS = Collections.unmodifiableList(Arrays.asList(
            new SettingsImportPreset("Suspended only", false, true, false, "", false, 7.0, 2, 1, false, ""),
            new SettingsImportPreset("Kani tag", false, false, true, "kani", false, 7.0, 2, 1, false, ""),
            new SettingsImportPreset("Leech tag", false, false, true, "leech", false, 7.0, 2, 1, false, ""),
            new SettingsImportPreset("Mining deck", false, false, false, "", false, 7.0, 2, 1, true, "deck:Mining"),
            new SettingsImportPreset("Recent fails", false, false, false, "", false, 7.0, 2, 1, true, "rated:30:1")
    ));

    public static List<SettingsImportPreset> defaults() {
        return DEFAULTS;
    }

    public static int boolFlag(boolean value) {
        return value ? 1 : 0;
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/SettingsImportPreset.kt
package dev.bee.kanjianki.core

import java.util.Collections

data class SettingsImportPreset(
    val label: String,
    val activeCards: Boolean,
    val suspendedCards: Boolean,
    val taggedCards: Boolean,
    val tags: String,
    val weakCards: Boolean,
    val weakDifficulty: Double,
    val weakLapses: Int,
    val minMatchingCards: Int,
    val browserQueryCards: Boolean,
    val browserQuery: String,
) {
    fun label(): String = label

    fun activeCards(): Boolean = activeCards

    fun suspendedCards(): Boolean = suspendedCards

    fun taggedCards(): Boolean = taggedCards

    fun tags(): String = tags

    fun weakCards(): Boolean = weakCards

    fun weakDifficulty(): Double = weakDifficulty

    fun weakLapses(): Int = weakLapses

    fun minMatchingCards(): Int = minMatchingCards

    fun browserQueryCards(): Boolean = browserQueryCards

    fun browserQuery(): String = browserQuery

    companion object {
        private val DEFAULTS: List<SettingsImportPreset> = Collections.unmodifiableList(
            listOf(
                SettingsImportPreset("Suspended only", false, true, false, "", false, 7.0, 2, 1, false, ""),
                SettingsImportPreset("Kani tag", false, false, true, "kani", false, 7.0, 2, 1, false, ""),
                SettingsImportPreset("Leech tag", false, false, true, "leech", false, 7.0, 2, 1, false, ""),
                SettingsImportPreset("Mining deck", false, false, false, "", false, 7.0, 2, 1, true, "deck:Mining"),
                SettingsImportPreset("Recent fails", false, false, false, "", false, 7.0, 2, 1, true, "rated:30:1"),
            )
        )

        @JvmStatic
        fun defaults(): List<SettingsImportPreset> = DEFAULTS

        @JvmStatic
        fun boolFlag(value: Boolean): Int = if (value) 1 else 0
    }
}
